estela-api/api/views/proxyprovider.py

```diff
@@ -0,0 +1,99 @@
+        proxy_attrs = [
+            "username",
+            "password",
+            "host",
+            "port",
+            "name",
+        ]
+        fields_and_values = vars(proxy_provider)
+        replaces = {
+            "password": "pass",
+            "host": "url",
+            "username": "user",
+        }
+        env_vars = []
+        for field, value in fields_and_values.items():
+            print(f"Field: {field}, Value: {value}")
+            if field in proxy_attrs:
+                name = replaces.get(field, field).upper()
+                print("Using the following name: %s" % name)
+                if name != "NAME": 
+                    masked = True
+                else:
+                    masked = False
+                env_vars.append({
+                    "name": f"ESTELA_PROXY_{name}",
+                    "value": value,
+                    "masked": masked
+                })
+        # if serializer.validated_data["level"] == "project":
+        #     env_vars_instance = SpiderJobEnvVar.objects.filter(project_id=serializer.validated_data["project_or_spider_id"])
+        # if serializer.validated_data["level"] == "spider":
+        #     env_vars_instance = SpiderJobEnvVar.objects.filter(spider_id=serializer.validated_data["project_or_spider_id"])
+        update_env_vars(instance, env_vars, level=serializer.validated_data["level"], delete=False)
+
+        if serializer.validated_data["level"] == "project":
+            env_vars_instance = SpiderJobEnvVar.objects.filter(project_id=serializer.validated_data["project_or_spider_id"])
+        if serializer.validated_data["level"] == "spider":
+            env_vars_instance = SpiderJobEnvVar.objects.filter(spider_id=serializer.validated_data["project_or_spider_id"])
+        env_vars_serialized = SpiderJobEnvVarSerializer(env_vars_instance, required=False, many=True)
+        resp_serializer = ProxyProviderResponseSerializer(
+            data={
+                "success": True,
+                "env_vars": env_vars_serialized.data
+            }
+        )
+        # response_ser = ProxyProviderResponseSerializer(data=rspse)
+        resp_serializer.is_valid()
+        return Response(resp_serializer.data, status=status.HTTP_200_OK)
+        #env_vars_serialized.is_valid(raise_exception=True)
+        import json
+        env_vars_json = json.dumps(env_vars_serialized.data)
+        print(env_vars_serialized.data)
+        return Response({
+            "success": True,
+            "env_vars": env_vars_json,
+        }, status=status.HTTP_200_OK, headers = {})
+        proxy_provider = ProxyProvider.objects.get(pk=pk)
+        print(proxy_provider)
+        serializer = ProxyProviderResponseSerializer(
+            instance, data=proxy_provider.__dict__, context=request.data,
+        )
+
+        serializer.is_valid(raise_exception=True)
+        headers = self.get_success_headers(serializer.data)
+        print(headers)
```

Comment:
Don't forget to remove unnecessary prints

Reply:
Done